src/boot/efi/stub.c

```diff
@@ -159,23 +162,74 @@ static void export_stub_variables(EFI_LOADED_IMAGE_PROTOCOL *loaded_image) {
         (void) efivar_set_str16(MAKE_GUID_PTR(LOADER), u"StubInfo", u"systemd-stub " GIT_VERSION, 0);
 
         (void) efivar_set_uint64_le(MAKE_GUID_PTR(LOADER), u"StubFeatures", stub_features, 0);
+
+        (void) efivar_set_uint64_str16(MAKE_GUID_PTR(LOADER), u"StubProfile", profile, 0);
+}
+
+static bool parse_profile_from_cmdline(char16_t **cmdline, unsigned *ret_profile) {
+        assert(cmdline);
+        assert(*cmdline);
+        assert(ret_profile);
+
+        const char16_t *p = *cmdline;
+        if (p[0] != '@')
+                goto nothing;
+
+        uint64_t u;
+        const char16_t *tail;
+        if (!parse_number16(p + 1, &u, &tail))
+                goto nothing;
+        if (u > UINT_MAX)
+                goto nothing;
+        if (tail[0] == u' ')
+                tail++;
+        else if (tail[0] != 0)
+                goto nothing;
```

Comment:
Not sure what this check is supposed to do?

Reply:
We allow two cases: The `@` profile prefix is prefixed to some kernel command line. or it's used all alone. This covers for the latter case. it just checks if we are at the end of string. if yes, great, it's a profile expression. If not, we assume it is something else.

Reply:
will add comment